scanner/src/main/kotlin/experimental/ExperimentalScanner.kt

```diff
@@ -541,17 +541,28 @@ class ExperimentalScanner(
             return
         }
 
-        val provenancesWithMissingArchives = nestedProvenances.filterNot { (_, nestedProvenance) ->
-            archiver.hasArchive(nestedProvenance.root)
-        }
+        val provenancesWithMissingArchives = controller.getNestedProvenancesByPackage()
+            .filterNot { (_, nestedProvenance) -> archiver.hasArchive(nestedProvenance.root) }
 
         log.info { "Creating file archives for ${provenancesWithMissingArchives.size} package(s)." }
 
         val duration = measureTime {
-            provenancesWithMissingArchives.forEach { (_, nestedProvenance) ->
-                val dir = downloadRecursively(nestedProvenance)
-                archiver.archive(dir, nestedProvenance.root)
-                dir.safeDeleteRecursively(force = true)
+            provenancesWithMissingArchives.forEach { (pkg, nestedProvenance) ->
+                runCatching {
+                    downloadRecursively(nestedProvenance)
+                }.onSuccess { dir ->
+                    archiver.archive(dir, nestedProvenance.root)
+                    dir.safeDeleteRecursively(force = true)
+                }.onFailure {
+                    controller.addProvenanceResolutionIssue(
```

Comment:
Provenance resolution issues happen only for packages which cannot be scanned. Therefore the implementation expects that no scan results exist for those packages and later on creates empty scan results with only these issues.
Using this function here would result in duplicate scan results for those packages.

Instead, I would recommend to the the folowing:
* Add a new `addIssue()` function to the `ScanController` backed by its own map.
* Add these issues to the scan results created in `getNestedScanResultsByPackage()`, probably best by accessing above map in `buildNestedProvenanceScanResult()`.